File: backend/JobTrail.Api/Models/Candidatura.cs
using System.ComponentModel.DataAnnotations;

namespace JobTrail.Api.Models
{
    public enum StatusCandidatura
    {
        Aplicado = 0,
        Triagem = 1,
        TesteTecnico = 2,
        Entrevista = 3,
        Oferta = 4,
        Rejeitado = 5,
        Desistiu = 6
    }

    public class Candidatura
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(150)]
        public string Cargo { get; set; } = string.Empty;

        [MaxLength(500)]
        public string? LinkVaga { get; set; }

        public decimal? SalarioPretendido { get; set; }

        public decimal? SalarioOferecido { get; set; }

        [MaxLength(1000)]
        public string? Notas { get; set; }

        public StatusCandidatura Status { get; set; } = StatusCandidatura.Aplicado;

        public DateTime DataAplicacao { get; set; } = DateTime.UtcNow;

        public DateTime CriadoEm { get; set; } = DateTime.UtcNow;

        // Relacionamento: cada candidatura pertence a uma empresa
        public int EmpresaId { get; set; }
        public Empresa? Empresa { get; set; }

        // Relacionamento: uma candidatura tem várias entradas de histórico
        public ICollection<HistoricoStatus> Historico { get; set; } = new List<HistoricoStatus>();
    }
}

File: frontend/src/types/candidatura.ts
// Espelha os DTOs do backend (JobTrail.Api/DTOs/CandidaturaDtos.cs).
// O enum StatusCandidatura chega como string por causa do JsonStringEnumConverter
// configurado no Program.cs (ex: "Entrevista", não 3).

export type StatusCandidatura =
  | 'Aplicado'
  | 'Triagem'
  | 'TesteTecnico'
  | 'Entrevista'
  | 'Oferta'
  | 'Rejeitado'
  | 'Desistiu'

export const STATUS_CANDIDATURA: StatusCandidatura[] = [
  'Aplicado',
  'Triagem',
  'TesteTecnico',
  'Entrevista',
  'Oferta',
  'Rejeitado',
  'Desistiu',
]

export const STATUS_LABEL: Record<StatusCandidatura, string> = {
  Aplicado: 'Aplicado',
  Triagem: 'Triagem',
  TesteTecnico: 'Teste técnico',
  Entrevista: 'Entrevista',
  Oferta: 'Oferta',
  Rejeitado: 'Rejeitado',
  Desistiu: 'Desistiu',
}

export interface Candidatura {
  id: number
  cargo: string
  empresaId: number
  nomeEmpresa: string
  linkVaga: string | null
  salarioPretendido: number | null
  salarioOferecido: number | null
  notas: string | null
  status: StatusCandidatura
  dataAplicacao: string
  criadoEm: string
}

// Espelha CandidaturaRequestDto — usado para CRIAR
export interface CandidaturaPayload {
  cargo: string
  empresaId: number
  linkVaga?: string | null
  salarioPretendido?: number | null
  salarioOferecido?: number | null
  notas?: string | null
  dataAplicacao?: string | null
}

// Espelha CandidaturaUpdateDto — usado para ATUALIZAR.
// Sem empresaId (a empresa não pode ser trocada por aqui) e sem status
// (mudança de status é feita pelo endpoint dedicado PATCH /status).
export interface CandidaturaUpdatePayload {
  cargo: string
  linkVaga?: string | null
  salarioPretendido?: number | null
  salarioOferecido?: number | null
  notas?: string | null
}

// Espelha HistoricoStatusResponseDto
export interface HistoricoStatus {
  id: number
  statusAnterior: StatusCandidatura
  statusNovo: StatusCandidatura
  dataMudanca: string
}
